### ISSUE TITLE
Reduce number of dependencies

### ISSUE THREAD

**Issue by brk00** [codles/EEG.jl#25]
Maybe we should make the package more lightweight and faster to load by removing some packages from the dependencies.

Docile will eventually drop after 0.4 release, and some other could be removed (like ProgressMeter) or made optional (via Requires.jl, like MAT, BDF, etc.). I also belive that importing Winston and Gadfly may be a little to much, and make our loadtime go too high.

By the way, maybe we should import packages only on `EEG.jl`, like most packages do.

**Comment by codles** [comment]
Agreed. I hadn't seen Requires.jl before.
Ill start by tidying up the importing of packages then start on reducing dependencies.

**Comment by codles** [comment]
Also two graphics packages feels like overkill.

**Comment by codles** [comment]
Im thinking of settling on just Gadfly. Any opinion @brk00 ?

**Comment by brk00** [comment]
Gadfly sounds good. It's slow right now on importing, but when precompile options arrive, it will become much faster. Maybe making plot functions available with `@require Gadfly`? They would only be exposed if the user import Gadfly explicitly.

Can you explain me how Require's `@lazymod` work? I can't figure it out.

**Comment by codles** [comment]
Good idea to put the plotting functions in a require. I have started converting the Winston functions to Gadfly. Once that's done I'll wrap it in `@require`. I will also check out lazymod this week and get back to you.
